Libs/Visualization/VTK/Widgets/ctkVTKChartView.cpp

```diff
@@ -135,53 +176,68 @@ void ctkVTKChartViewPrivate::chartBounds(double* bounds)const
   Q_Q(const ctkVTKChartView);
   bounds[0] = bounds[2] = bounds[4] = bounds[6] = VTK_DOUBLE_MAX;
   bounds[1] = bounds[3] = bounds[5] = bounds[7] = VTK_DOUBLE_MIN;
-  vtkChartXY* chart = q->chart();
+  vtkChart* chart = q->abstractChart();
+  vtkChartXY* chartXY = vtkChartXY::SafeDownCast(chart);
   const vtkIdType plotCount = chart->GetNumberOfPlots();
   for (vtkIdType i = 0; i < plotCount; ++i)
     {
     vtkPlot* plot = chart->GetPlot(i);
 
-    int corner = chart->GetPlotCorner(plot);
-    double plotBounds[4];
-    plot->GetBounds(plotBounds);
-    switch (corner)
+    if (chartXY)
       {
-      // bottom left
-      case 0:
+      int corner = chartXY->GetPlotCorner(plot);
+      double plotBounds[4];
+      plot->GetBounds(plotBounds);
+      switch (corner)
+        {
+        // bottom left
+        case 0:
+          // x
+          bounds[2] = bounds[2] > plotBounds[0] ? plotBounds[0] : bounds[2];
```

Comment:
would be more readable with std::min and std::max

Reply:
as decided to keep similar as the above code for consistency.